### ISSUE TITLE
iRule regex parsing might cause an indefinite hang

### ISSUE THREAD

**Issue by jszczepanski** [f5devcentral/f5-automation-config-converter#78]
### Environment
 * ACC Version: 1.18
 * AS3 Version: N/A
 * BIG-IP Version: N/A

### Summary
Found in a client UCS archive.

Parsing the following conf file causes ACC to become unresponsive, likely stuck in an infinite loop with 100% CPU usage. The freeze seems to only happen when there's another `ltm rule` object somewhere after the one with this problematic regex, otherwise we get a regular exception.

### Steps To Reproduce
1. Convert following config:
```conf
ltm rule /tenant_268/application_268/l4j {
when HTTP_REQUEST_DATA {
  set tmpPayload [HTTP::payload]
  set payload [URI::decode $tmpPayload]
  while { $payload ne $tmpPayload } {
    set tmpPayload $payload
    set payload [URI::decode $tmpPayload]
  }

  if {[string tolower $payload] matches_regex {\$\{\s*(j|\$\{).+?\}}} {
    drop
    event disable all
  }
}
}
ltm rule /tenant_268/application_268/foo {

}
```

2. ACC is now frozen indefinitely.

Note that if we attempt to convert the first iRule without including the second one (or replacing it with any object, e.g. `ltm pool`), we get the following error:

```
/app/src/engines/parser.js:150
                if (!(line.trim().startsWith('#') && ruleFlag)) {
                           ^

TypeError: Error parsing input file. Please open an issue at https://github.com/f5devcentral/f5-automation-config-converter/issues and include the following error:
Cannot read properties of undefined (reading 'trim')
    at groupObjects (/app/src/engines/parser.js:150:28)
    at /app/src/engines/parser.js:255:30
    at Array.forEach (<anonymous>)
    at module.exports (/app/src/engines/parser.js:184:28)
    at mainRunner (/app/src/main.js:39:18)
    at main (/app/src/main.js:128:30)
```

**Comment by delgadillo22** [comment]
Thank you for the report, I've added this to our internal backlog as CHARON-584 and will get this prioritized.

**Comment by p-semenov-f5** [comment]
Fixed in 1.19
